### PR TITLE
LUT-27671 : no deletion of fields already filled in and deactivation

### PR DESCRIPTION
**Pull request by obfromevi**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -72,6 +72,17 @@
             <type>lutece-plugin</type>
             <version>[1.0.0,)</version>
         </dependency>
+        <dependency>
```
Comment (seboo) [inline pom.xml:75]: This module already depends on the forms plugin, this could create circular dependency issues

**Review by seboo — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (obfromevi) [reply pom.xml]: the unnecessary dependencies has been removed

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java [hunk 2]
```diff
@@ -560,6 +582,51 @@ public String getConfirmRemoveComposite( HttpServletRequest request )
...
+        	    isCompositeHasResponses = isExisteQuestionFilled( responsefilter );
+            }
+        }
+    	
+    	return isCompositeHasResponses;
+    }
+    
+    private boolean isExisteQuestionFilled( ResponseFilter responsefilter )
+    {
+    	boolean isExisteQuestionFilled = false ;
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java:618]: rename to existsFilledResponse

File: src/java/fr/paris/lutece/plugins/forms/service/FormService.java [hunk 3]
```diff
@@ -604,4 +606,23 @@ public void saveOrUpdateField( Entry entry, String fieldName, String title, Stri
...
     }
+
+    /**
+     * Get the responses which verify the filter
+     * 
+     * @param responseFilter
+     *            the filter
+     * @return the list of responses
+     */
+    public List<Response> getResponseList( ResponseFilter responseFilter )
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/forms/service/FormService.java:617]: unnecessary method

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java [hunk 4]
```diff
@@ -560,6 +582,51 @@ public String getConfirmRemoveComposite( HttpServletRequest request )
...
+    	
+    	List<Response> responseList = _formService.getResponseList( responsefilter );
```
Comment (seboo) [inline src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java:620]: Use directly  ResponseHome.getResponseList( responseFilter );
(FormHome and StepHome are already used)

maybe the instance variable _form should be used ? (without forgetting to test that it is still  the right form)

**Review by seboo — CHANGES_REQUESTED** [review]

File: src/java/fr/paris/lutece/plugins/forms/web/CompositeQuestionDisplay.java [hunk 5]
```diff
@@ -281,6 +283,46 @@ public String getCompositeHtml( HttpServletRequest request, List<FormQuestionRes
 
         return strQuestionTemplate;
     }
+    
+    private boolean isCompositeEnabled( Entry entry, List<FormQuestionResponse> listFormQuestionResponse, DisplayType displayType )
```
[the remark this answers is not in the document]
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/CompositeQuestionDisplay.java]: The method has been renamed and comments added for more clarity

File: src/java/fr/paris/lutece/plugins/forms/web/CompositeQuestionDisplay.java [hunk 6]
```diff
@@ -281,6 +283,46 @@ public String getCompositeHtml( HttpServletRequest request, List<FormQuestionRes
...
+                	isCompositeEnabled = isResponseValueExiste( listFormQuestionResponse );
+                    break;
+                default: // Nothing to do
+            }
+        }
+
+        return isCompositeEnabled;
+    }
+    
+    private boolean isResponseValueExiste( List<FormQuestionResponse> listFormQuestionResponse )
```
[the remark this answers is not in the document]
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/CompositeQuestionDisplay.java]: the change requested has been implemented

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java [hunk 7]
```diff
@@ -560,6 +582,51 @@ public String getConfirmRemoveComposite( HttpServletRequest request )
...
+        	    responsefilter.setListIdEntry( idEntryList );
+        		
+        	    isCompositeHasResponses = isExisteQuestionFilled( responsefilter );
+            }
+        }
+    	
+    	return isCompositeHasResponses;
+    }
+    
+    private boolean isExisteQuestionFilled( ResponseFilter responsefilter )
```
[the remark this answers is not in the document]
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java]: the change requested has been implemented

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java [hunk 8]
```diff
@@ -722,6 +789,64 @@ public String doMoveComposite( HttpServletRequest request )
...
+    
+    /**
+     * Modify the field attribute disabled
+     * 
+     * @param request
+     *            The HTTP request
+     * @return The URL to go after performing the action
+     */
+    @Action( value = ACTION_MODIFY_FIELD_DISABLED )
+    public String doModifyFieldDisabled( HttpServletRequest request )
```
[the remark this answers is not in the document]
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java]: the change requested has been implemented

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java [hunk 9]
```diff
@@ -722,6 +789,64 @@ public String doMoveComposite( HttpServletRequest request )
...
+
+        if ( nIdQuestion != INTEGER_MINUS_ONE )
+        {
+            Question questionToUpdate = QuestionHome.findByPrimaryKey( nIdQuestion );
+            Entry entry = questionToUpdate.getEntry( );	
+        	
+            if ( entry != null )
+            {
+                Field disabledField = entry.getFieldByCode( IEntryTypeService.FIELD_DISABLED );
+                boolean valueField = disabledField != null ? !Boolean.parseBoolean( disabledField.getValue( ) ) : false;
```
[the remark this answers is not in the document]
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java]: the change requested has been implemented

File: src/java/fr/paris/lutece/plugins/forms/web/admin/AbstractFormQuestionJspBean.java [hunk 10]
```diff
@@ -97,6 +97,7 @@ public abstract class AbstractFormQuestionJspBean extends AbstractJspBean
     protected static final String ACTION_MODIFY_GROUP = "modifyGroup";
     protected static final String ACTION_DUPLICATE_QUESTION = "duplicateQuestion";
     protected static final String ACTION_SAVE_QUESTION = "saveQuestion";
+    protected static final String ACTION_MODIFY_FIELD_DISABLED = "modifyFieldDisabled";
```
[the remark this answers is not in the document]
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/admin/AbstractFormQuestionJspBean.java]: the change requested has been implemented

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java [hunk 11]
```diff
@@ -722,6 +789,64 @@ public String doMoveComposite( HttpServletRequest request )
...
+    	int nIdQuestion = NumberUtils.toInt( request.getParameter( FormsConstants.PARAMETER_ID_QUESTION ), INTEGER_MINUS_ONE );
+
+        if ( nIdQuestion != INTEGER_MINUS_ONE )
+        {
+            Question questionToUpdate = QuestionHome.findByPrimaryKey( nIdQuestion );
+            Entry entry = questionToUpdate.getEntry( );	
+        	
+            if ( entry != null )
+            {
+                Field disabledField = entry.getFieldByCode( IEntryTypeService.FIELD_DISABLED );
```
[the remark this answers is not in the document]
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java]: the change requested has been implemented

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java [hunk 12]
```diff
@@ -107,13 +114,18 @@ public class FormQuestionJspBean extends AbstractFormQuestionJspBean
...
+    private static final String MESSAGE_CANT_REMOVE_ENTRY_RESOURCES_ATTACHED = "forms.message.cantRemoveEntry.resourceAttached";
+    private static final String MESSAGE_CANT_REMOVE_GROUP_RESOURCES_ATTACHED = "forms.message.cantRemoveGroup.resourceAttached";
 
     // Warning messages
     private static final String WARNING_CONFIRM_REMOVE_QUESTION_FORM_ACTIVE = "forms.warning.deleteComposite.confirmRemoveQuestion.formActive";
     private static final String WARNING_CONFIRM_REMOVE_GROUP_ANY_QUESTIONS_FORM_ACTIVE = "forms.warning.deleteComposite.confirmRemoveGroup.formActive";
 
     // Markers
     private static final String MARK_FIELDS_LIST_BY_ID_ENTRIES = "fields_list_by_id_entries";
+    private static final String MARK_ID_QUESTION_ENTRY_MAP = "idQuestionEntryMap";
```
[the remark this answers is not in the document]
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java]: the change requested has been implemented

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java [hunk 13]
```diff
@@ -155,6 +167,7 @@ public String getManageQuestions( HttpServletRequest request )
         model.put( FormsConstants.MARK_ENTRY_TYPE_REF_LIST, FormsEntryUtils.initListEntryType( ) );
         model.put( FormsConstants.MARK_ID_PARENT, _nIdParentSelected );
         model.put( SecurityTokenService.MARK_TOKEN, SecurityTokenService.getInstance( ).getToken( request, ACTION_SAVE_QUESTION ) );
+        model.put( MARK_ID_QUESTION_ENTRY_MAP, getIdQuestionEntryMap( listICompositeDisplay ) );
```
[the remark this answers is not in the document]
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java]: it is used to get the entry id and the deactivation status in order to display the appropriate button

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java [hunk 14]
```diff
@@ -560,6 +582,51 @@ public String getConfirmRemoveComposite( HttpServletRequest request )
 
         return redirect( request, strMessageUrl );
     }
+    
+    private boolean isCompositeHasResponses( int nIdEntry )
```
[the remark this answers is not in the document]
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java]: The change requested has been implemented but in my opinion the first level may cause a performance issue.
Do you still want to keep this level or did you have another idea than the one I thought of ?

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java (hunk 2 above)
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java]: The change requested has been implemented

File: src/java/fr/paris/lutece/plugins/forms/service/FormService.java (hunk 3 above)
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/service/FormService.java]: The change requested has been implemented

File: src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java (hunk 4 above)
Comment (obfromevi) [reply src/java/fr/paris/lutece/plugins/forms/web/admin/FormQuestionJspBean.java]: The change requested has been implemented

**Pull request merged**